src/rust/src/contract.rs

```diff
@@ -209,30 +209,17 @@ fn encode_diagnostic_events(events: &Vec<DiagnosticEvent>) -> Vec<RustBuf> {
 
 fn extract_ledger_effects(
     entry_changes: Vec<LedgerEntryChange>,
-) -> (Vec<RustBuf>, Vec<ReadOnlyBump>) {
+) -> Vec<RustBuf> {
     let mut modified_entries = vec![];
-    let mut read_only_bumps = vec![];
 
     for change in entry_changes {
-        if change.read_only {
-            // Only return expiration bumps for read-only entries (if any).
-            if let Some(expiration_change) = change.expiration_change {
-                if expiration_change.new_expiration_ledger > expiration_change.old_expiration_ledger
-                {
-                    read_only_bumps.push(ReadOnlyBump {
-                        ledger_key: change.encoded_key.into(),
-                        min_expiration: expiration_change.new_expiration_ledger,
-                    });
-                }
-            }
-        } else {
+        assert!(!change.read_only);
             if let Some(encoded_new_value) = change.encoded_new_value {
                 modified_entries.push(encoded_new_value.into());
             }
-        }
     }
 
-    (modified_entries, read_only_bumps)
+    modified_entries
```

Comment:
I think this function should be converting `LedgerEntryChange` to (maybe) a LE change and (maybe) an EE change. But that really depends on the host-side implementation at this point.